floodviz/views.py

```diff
@@ -1,7 +1,79 @@
-from flask import request, render_template, abort
+from flask import request, render_template, abort #, jsonify
+
+## below is for data processing ##
+import json
+import time
+from datetime import datetime
+import requests
+import operator
+# # # # # # # # # # # # # # # # # #
 
 from . import app
 
 @app.route('/home/')
 def home():
-    return render_template('index.html')
\ No newline at end of file
+  	## new data store ##
+	newd = []
+	sites_value_maxs = {}
+	sites = app.config['SITE_IDS'] 
+	start_date = app.config['START_DT']
+	end_date = app.config['END_DT']
+	n_show_series = app.config['N_SERIES']
+	hydro_meta = app.config['HYDRO_META']
+
+	## Set up to retrieve all site ids from url? ##
+	for idx, site in enumerate(sites):
+		url = 'https://nwis.waterservices.usgs.gov/nwis/iv/?site=' + site + '&startDT=' + start_date + '&endDT=' + end_date + '&parameterCD=00060&format=json'
+
+		try:
+			r = requests.get(url)
+		except:
+			print('Unable to retrieve URL: ' + url)
+			continue
+		
+
+		if r.status_code is 200:
+			j = r.json()
+			siteName = j['value']['timeSeries'][0]['sourceInfo']['siteName']
+			key = siteName # Key for this series
+			newd.append({'key': key, 'values': [], 'max': 0})
+			# Fill new data
+			max_val = 0
+			for idx2, obj in enumerate(j['value']['timeSeries'][0]['values'][0]['value']):
+				value = obj['value']
+				# For filtering series #
+				if max_val < float(value):
+					max_val = float(value)
+				#sum_values += int(value)
+				#num_values += 1
+				dt = obj['dateTime']
+				date = dt.split('T')[0]
+				t = dt.split('T')[1].split('.')[0]
+			 	# reformat datetime for python datetime
+				dt = datetime.strptime(date + ' ' + t, '%Y-%m-%d %H:%M:%S')
+				# Conver to miliseconds for use with d3 x axis format
+				dt_ms = time.mktime(dt.timetuple()) * 1000
+				# create dummy value for nvd3 issue at https://github.com/novus/nvd3/issues/695 #
+				if idx2 is 0:
+					newd[idx]['values'].append({'date':start_date, "time": 0, "time_mili": dt_ms, 'value': 0, 'max': max_val})
+				# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
```

Comment:
There is an issue within NVD3 and cumulativeLineCharts that normalizes the Y axis data values incorrectly, thus misrepresenting them on the Y axis. In the link in the comment, NatasaPeic presents a workaround in which you create a dummy data value of 0 for only the first data point. After putting that in, the graph renders correctly.